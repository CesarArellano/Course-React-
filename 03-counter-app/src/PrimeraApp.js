//import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

// Functional Components

const PrimeraApp = ({ saludo, subtitulo }) => {
  // const saludo = 'Hola Mundo const';
  // const objeto = {
  //   nombre: 'César',
  //   edad: 21
  // };
  // console.log(props);

  // if(!saludo) {
  //   throw new Error('El saludo es necesario');
  // }
  return (
    // <Fragment>
    //   <h1>Hola Mundo</h1>
    //   <p>Hola</p>
    // </Fragment>
    <>
      <h1>{ saludo }</h1>
      {/* <pre>{ JSON.stringify(objeto, null, 2)}</pre> */}
      <p>{ subtitulo }</p>
    </>    
  );
}

PrimeraApp.propTypes = {
  saludo: PropTypes.string.isRequired,
  subtitulo: PropTypes.string
}

PrimeraApp.defaultProps = {
  subtitulo: 'Soy un subtítulo'
}

export default PrimeraApp;