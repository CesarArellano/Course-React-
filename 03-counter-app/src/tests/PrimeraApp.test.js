import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
// import { render } from '@testing-library/react';

import PrimeraApp from '../PrimeraApp';

describe('Tests in <PrimerApp/>', () => {
  // test('Must show the message "Hola, Soy Gokú"', () => {
  //   const greetings = 'Hola, soy Gokú';
  //   const { getByText } = render(<PrimeraApp saludo={ greetings } />);
  //   expect( getByText(greetings) ).toBeInTheDocument();
  // })

  test('Must be show <PrimerApp /> correctly', () => {
    const saludo = 'Hola, soy César';
    const wrapper = shallow(<PrimeraApp saludo={ saludo } />)
    expect(wrapper).toMatchSnapshot();
  });
});
