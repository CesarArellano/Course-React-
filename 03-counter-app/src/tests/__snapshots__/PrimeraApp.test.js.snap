// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <PrimerApp/> Must be show <PrimerApp /> correctly 1`] = `
<Fragment>
  <h1>
    Hola, soy César
  </h1>
  <p>
    Soy un subtítulo
  </p>
</Fragment>
`;
